backstop/tests/carousel.html

```diff
@@ -428,9 +410,16 @@ <h2>Default</h2>
         prev.addEventListener('click', () => goPrevSlide());
         next.addEventListener('click', () => goNextSlide());
 
+        // update dots truncation on resize
+        const resizeObserver = new ResizeObserver(([{ contentRect: { width } }]) => {
+          MAX_DOTS = Math.floor(width / dots[0].offsetWidth);
+          updateSlide(currentSlide);
+        });
+        resizeObserver.observe(dotsList);
```

Comment:
If you find a clean solution for avoiding clipping of dots while keeping the smooth scrolling animation working as intended, I'd be happy to implement it.

Reply:
For more context:

_Previously_, the clipping dots were a non-issue from a CSS perspective because we were planning to use resize observer to apply `iui-invisible` on any dots that wouldn't fit.

_Now_, we can't really apply `iui-invisible` because the animation (i.e. smooth scrolling) requires the dots to be already present when they are entering the view from the edges.

Reply:
Would the "Resize observer" you mentioned be able to prevent the clipping of the dots?  So there would be no clipping in React but there will be clipping in CSS?

Reply:
How? Resize observer is not magic, it just lets you observe size changes. My current html demo already makes use of it.

Reply:
I have not worked with resize observer, I don't know what it is capable of.  If we say each dot button is equal to 40px wide, I was thinking resize observer could watch the width of the parent.  If the parent was 410px wide for example, it would shrink the width to 400px so that there is no clipping.

Reply:
I am fine with this PR besides this clipping issue and if this is causing a clog for the react stuff I can resolve and approve.